.ci/travis_before_script.sh

```diff
@@ -24,7 +24,7 @@ if [[ "${USE_CMAKE:-no}" == "yes" ]] &&  [[ "${USE_CONAN:-no}" == "no" ]]; then
     mkdir -p $BUILD_DIR
     pushd $BUILD_DIR
     cmake $SRC_DIR -DCMAKE_BUILD_TYPE=Debug \
-                   -DCMAKE_PREFIX_PATH=$HDF5_DIR/share/cmake \
+                   -DHDF5_ROOT=$HDF5_DIR \
```

Comment:
Can you revert this as well? So CI uses the targets specified by `hdf5-config`.

Reply:
I think HDF5 uses hdf5-config.
See the latest item at https://cmake.org/cmake/help/latest/module/FindHDF5.html

Reply:
Yes, but it needs to know where to find `hdf5-config`. That is wat `CMAKE_PREFIX_PATH=$HDF5_DIR/share/cmake` is for. Now it just searches for `libhdf5.so` in `HDF5_ROOT`.

You can also see the difference in travis output. With `hdf5-config`:
```
--- Found HDF5: hdf5::hdf5-shared (found version "1.10.6")
```
Without:
```
-- Found HDF5: /home/travis/build/tbeu/matio/hdf5-1.10.7/hdf5/lib/libhdf5.so (found suitable version "1.10.7", minimum required is "1.8")  
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found suitable version "1.2.8", minimum required is "1.2.3") 
```